### PR TITLE
docs: Added FCN to the new doc

### PR DESCRIPTION
**Pull request by frgfm**
Following up on https://github.com/pytorch/vision/issues/5897, this PR introduces the following modifications:

- creates the new rst doc file for FCN
- adds the reference on the new page
- updates the docstring accordingly

Any feedback is welcome!

cc @NicolasHug @datumbox

### CODE REVIEW

**Review by NicolasHug — APPROVED** [review]
Thanks @frgfm !!

File: torchvision/models/segmentation/fcn.py [hunk 1]
```diff
@@ -115,15 +115,26 @@ def fcn_resnet50(
...
+            pretrained weights to use. See
+            :class:`~torchvision.models.segmentation.FCN_ResNet50_Weights` below for
+            more details, and possible values. By default, no pre-trained
+            weights are used.
+        progress (bool, optional): If True, displays a progress bar of the
+            download to stderr. Default is True.
+        **kwargs: parameters passed to the ``torchvision.models.segmentation.fcn.FCN``
+            base class. Please refer to the `source code
+            <https://github.com/pytorch/vision/blob/main/torchvision/models/segmentation/fcn.py>`_
+            for more details about this class.
```
Comment (NicolasHug) [inline torchvision/models/segmentation/fcn.py:132]: Ah, I approved a bit fast. I thought the CI issue was just linting but it looks like some parameters aren't documented (aux_loss, etc.): https://app.circleci.com/pipelines/github/pytorch/vision/17141/workflows/a7ac687b-9bdd-40cb-a929-f8094fc88b5d/jobs/1389276

Would you mind addressing these @frgfm ?

File: docs/source/models/fcn.rst [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+Segmentation <https://arxiv.org/abs/1411.4038>`__
+paper.
+
+
+Model builders
+--------------
+
+The following model builders can be used to instantiate a FCN model, with or
+without pre-trained weights. All the model builders internally rely on the
+``torchvision.models.segmentation.fcn.FCN`` base class. Please refer to the `source
```
Comment (oke-aditya) [inline docs/source/models/fcn.rst:16]:
```suggestion
``torchvision.models.segmentation.FCN`` base class. Please refer to the `source
```

Comment (oke-aditya) [reply docs/source/models/fcn.rst:16]: Should we encourage to use module import than importing from fcn.py file?

File: torchvision/models/segmentation/fcn.py [hunk 3]
```diff
@@ -115,15 +115,30 @@ def fcn_resnet50(
...
+        weights (:class:`~torchvision.models.segmentation.FCN_ResNet50_Weights`, optional): The
+            pretrained weights to use. See
+            :class:`~torchvision.models.segmentation.FCN_ResNet50_Weights` below for
+            more details, and possible values. By default, no pre-trained
+            weights are used.
+        progress (bool, optional): If True, displays a progress bar of the
+            download to stderr. Default is True.
+        num_classes (int, optional): number of output classes of the model (including the background).
+        aux_loss (bool, optional): If True, it uses an auxiliary loss.
+        weights_backbone (:class:`~torchvision.models.ResNet50_Weights, optional): The pretrained
```
Comment (NicolasHug) [inline torchvision/models/segmentation/fcn.py:131]:
```suggestion
        weights_backbone (:class:`~torchvision.models.ResNet50_Weights`, optional): The pretrained
```

File: torchvision/models/segmentation/fcn.py [hunk 4]
```diff
@@ -156,15 +171,30 @@ def fcn_resnet101(
...
+        weights (:class:`~torchvision.models.segmentation.FCN_ResNet101_Weights`, optional): The
+            pretrained weights to use. See
+            :class:`~torchvision.models.segmentation.FCN_ResNet101_Weights` below for
+            more details, and possible values. By default, no pre-trained
+            weights are used.
+        progress (bool, optional): If True, displays a progress bar of the
+            download to stderr. Default is True.
+        num_classes (int, optional): number of output classes of the model (including the background).
+        aux_loss (bool, optional): If True, it uses an auxiliary loss.
+        weights_backbone (:class:`~torchvision.models.ResNet101_Weights, optional): The pretrained
```
Comment (NicolasHug) [inline torchvision/models/segmentation/fcn.py:187]:
```suggestion
        weights_backbone (:class:`~torchvision.models.ResNet101_Weights`, optional): The pretrained
```

**Review by NicolasHug — APPROVED** [review]
Thanks @frgfm !

**Pull request merged**